### DISCUSSION TITLE
How to run Avalonia-Window in library?

### DISCUSSION THREAD

**Discussion by copycd** [AvaloniaUI/Avalonia#8219] [Q&A] [answered]
I have an existing project that consists of .net6-window that is usual .net winform app.
I want to show avalonia window using another .net6 dll-project.
therefore I inserted a avalonia Window in the dll-project and called the window in Main-programe.
but, An error occurred.

![image](https://user-images.githubusercontent.com/2500325/170858971-f4a819fe-8402-4c6b-bf03-1f4b0d46d3b9.png)

**Comment by kekekeks** [comment, marked as answer]
You need to properly configure the framework before using it. Otherwise Avalonia doesn't know which platform backend it should use, which rendering backend it should use, which theme it should use, etc, etc. 

You also need to run the main loop on **the first thread of the application** (the one where `Main` was called), that's a limitation of macOS.
